src/lib/api/contrib/invenio-records-resources/InvenioRecordsResourcesRequestSerializer.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of React-SearchKit.
+ * Copyright (C) 2021 CERN.
+ *
+ * React-SearchKit is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import Qs from 'qs';
+import _extend from 'lodash/extend';
+import _isEmpty from 'lodash/isEmpty';
+
+/** Default backend request serializer for invenio-records-resources */
+export class InvenioRecordsResourcesRequestSerializer {
+  constructor() {
+    this.serialize = this.serialize.bind(this);
+  }
+
+  _checkFilter = (filter) => {
+    if (!Array.isArray(filter)) {
+      throw new Error(
+        `Filter value "${filter}" in query state must be an array.`
+      );
+    }
+    if (!(filter.length === 2 || filter.length === 3)) {
+      throw new Error(
+        `Filter value "${filter}" in query state must be an array of 2 or 3 elements`
+      );
+    }
+  }
+
+  _addFilter = (filter, filterUrlParams) => {
+    this._checkFilter(filter)
+    const aggName = filter[0];
+    const fieldValue = filter[1];
+  	const hasChild = filter.length === 3;
+    if (hasChild) {
+      this._checkFilter(filter[2])
+    }
+    if (aggName in filterUrlParams) {
+      if (hasChild) {
+        filterUrlParams[aggName].push(fieldValue + "::" + filter[2][1])
```

Comment:
I would make this `::` parametrizable if possible.

Generally speaking, I was trying to avoid this separators and let `Qs` to encode the string correctly. E.g. if you have lists, then it will be automatically encoded in some standard way, and decoded in the same way in the 
backend.
For example, you could have `['report', 'pdf']` instead of `report::pdf` and then see how Qs encodes it. In this way you don't need separators and you don't have the prob that maybe you have a value (very unlikely) `rep::ort`.

Reply:
I tried with pushing as list instead of separated by `::`, 

- the `filterUrlParams` looks like this:

```
resource_type: (3) […]
​​
0: Array [ "image", "image-photo" ]
​​
1: "software"
​​
2: Array [ "publication", "publication-thesis" ]
```

- the `q` looks like this:

```
q=&sort=newest&page=1&size=10&resource_type=image&resource_type=image-photo&resource_type=software&resource_type=publication&resource_type=publication-thesis
```

​​I am afraid we need to keep the separator since `image-photo`, `publication-thesis` are not resource type but `image::image-photo` and `publication::publication-thesisis` are resource_type with sub_type.

Another approach that I was thinking is passing the `splitchar` from the config but I then I have to access config here.

Please let me know which one sounds good.

Reply:
Thanks for giving it a try, keep it like this then :)